fdbserver/DataDistribution.actor.cpp

```diff
@@ -3099,7 +3103,7 @@ ACTOR Future<Void> storageServerFailureTracker(
 		choose {
 			when ( wait(healthChanged) ) {
 				status->isFailed = !status->isFailed;
-				if(!status->isFailed && !server->teams.size()) {
+				if(!status->isFailed && (server->teams.size() < SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER || self->lastBuildTeamsFailed)) {
```

Comment:
The condition for the `server->teams.size()`  is not consistent with the team-number-per-server condition used in `addTeamsBestOf()`, which keeps building teams until each server reaches the `int targetTeamNumPerServer = (SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER * (configuration.storageTeamSize + 1)) / 2;`. 

Maybe we should use `server->teams.size() < targetTeamNumPerServer `, instead of `server->teams.size() < SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER`.

Reply:
I probably is not worth it, but you could submit a separate PR if you feel strongly about it

Reply:
I don't feel strongly about it, but I will submit a separate PR to get this out of my mind.